### DISCUSSION TITLE
Statistics API

### DISCUSSION THREAD

**Discussion by jxnxsdev** [tidal-music/discussions#32] [Ideas & Feature requests]
As far as I've seen, Tidal stores certain statistics. For example, it tracks how often a track has been listened to (when visiting an artist's profile, the tracks are mostly ordered by popularity) or what your top songs are. It would be nice to include the number of times a song has been played when requesting information about a song via the API. Additionally, there could be a general endpoint where you can submit one or multiple track IDs, and it provides statistics about the tracks. The same thing could be done with Artists.

From what I've read in other discussions, an interface via the API for a user's profile is also planned, where statistics could be added, such as the most played songs, activity, etc., depending on what is stored by Tidal.

**Comment by stefanmoro** [comment]
Hey @hdgamer1404Jonas,

Access to various stats is something we have been discussing if/how we can expose. It has also been requested/discussed earlier in e.g. #10, which contains more inforrmation.

**Comment by davistardif** [comment]
bump 

I have seen a "popularity" score in the official API docs (e.g. [here](https://developer.tidal.com/apiref?spec=catalogue&ref=get-track) in the get track response) but haven't been able to actually surface it. 

It's also referenced in the [openAPI file](https://developer.tidal.com/apiref/api-specifications/api-public-catalogue/tidal-catalog-openapi-3.0.json):
```
"popularity" : {
            "type" : "number",
            "description" : "Artist popularity (ranged in 0.00 ... 1.00). Conditionally visible",
            "format" : "double",
            "example" : 0.56
          }
``` 

Any idea what "conditionally visible" means here?

**Reply by stefanmoro** [reply]
Hey @davistardif ,

Our goal is to build one API to serve both our own 1st-party apps as well as third party apps. As a result of this goal, there may be some fields that we can only export to our 1st party apps (for various reasons). As of today, popularity is one such field.

As we work towards a V1 for TIDAL Developer, we hope to be able to expose more such conditional fields for 3rd party developers as well.

**Comment by davistardif** [comment]
in any case, looks like the popularity scores are easily scrapable

created a firefox extension as proof of concept: https://github.com/davistardif/tidal-popularity
